### PR TITLE
Tag Plots v0.5.0

### PR DESCRIPTION
**Pull request by tbreloff**
No longer supporting Julia v0.3.

Many fixes and features, including:
- `with` function for temporary defaults
- contours
- basic 3D plotting
- preliminary support for Bokeh
- `stroke` and `brush` for more fine-tuned control over visuals
- smarter "magic" arguments: `line`, `marker`

### CODE REVIEW

**Comment by Evizero** [comment]
nice! could you add a 3d plot example to the examples? I was actually thinking today that I would like to use your animation capabilities for a presentation of mine to generate a GIF of gradient descent etc walking down some error surfaces

**Comment by tbreloff** [comment]
I'll add examples eventually, but in the meantime you can follow
https://github.com/tbreloff/Plots.jl/issues/57.  Post there if there are
any 3D-specific requests/questions/bugs.
